### PR TITLE
ClientMiddleware: remove redundant operations

### PR DESCRIPTION
**Pull request by iRevive**
The clean-up is partially based on the changes from https://github.com/http4s/http4s-otel4s-middleware/pull/113.

### CODE REVIEW

File: core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala [hunk 1]
```diff
@@ -151,36 +148,19 @@ object ClientMiddleware {
...
-                      .eval {
-                        span.addAttributes(
-                          response(resp, allowedResponseHeaders) ++
-                            additionalResponseAttributes(resp.responsePrelude)
-                        )
-                      }
-                      .mapK(trace)
-                  }
-                case Outcome.Errored(e) =>
-                  Resource.eval(span.recordException(e)).mapK(trace)
```
Comment (iRevive) [inline core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala:168]: `Tracer[F].(...).build.resource` already sets the status upon finalization of a resource.

File: core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala [hunk 2]
```diff
@@ -151,36 +148,19 @@ object ClientMiddleware {
...
-                case Outcome.Canceled() =>
-                  // Canceled isn't always an error, but it generally is for http
-                  // TODO: decide if this should add "error", we do for the server side.
-                  Resource.eval(span.addAttribute(CustomAttributes.Canceled(true))).mapK(trace)
-              }) >> Resource.eval(span.addAttribute(CustomAttributes.exitCase(outcome))).mapK(trace)
+
+            _ <- Resource.onFinalizeCase { ec =>
+              span.addAttribute(CustomAttributes.exitCase(ec.toOutcome))
```
Comment (iRevive) [inline core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala:153]: `Span` has a [status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status) in the OpenTelemetry.  So `exit.case` looks like a duplication to me. 
Perhaps we can drop this explicit attribute?

File: core/src/test/scala/org/http4s/otel4s/middleware/ClientMiddlewareTests.scala [hunk 3]
```diff
@@ -89,4 +99,137 @@ class ClientMiddlewareTests extends CatsEffectSuite {
...
+
+            for {
+              f <- tracedClient.run(request).use_.start
+              _ <- f.joinWithUnit
+              spans <- testkit.finishedSpans
+            } yield {
+              val attributes = Attributes(spans.flatMap(_.attributes): _*)
+              assertEquals(attributes.get[String]("exit.case").map(_.value), Some("canceled"))
+              assertEquals(spans.flatMap(_.events), Nil)
+              assertEquals(spans.map(_.status), List(status))
```
Comment (iRevive) [inline core/src/test/scala/org/http4s/otel4s/middleware/ClientMiddlewareTests.scala:161]: See, `exit.case` attribute and `span.status` have the same meaning.

File: core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala (hunk 2 above)
Comment (NthPortal) [reply core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala:153]: let's go for it. there seems to be a lot of vestigial stuff from the migration (which is my fault 😅)

Comment (NthPortal) [reply core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala:153]: do you want to update this PR to do that here, or do it in a separate PR?

**Review by NthPortal — APPROVED** [review]

File: core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala (hunk 2 above)
Comment (iRevive) [reply core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala:155]: I will make a follow up PR.

**Pull request merged**